generators/servicehook/index.js

```diff
@@ -0,0 +1,258 @@
+﻿'use strict';
+const path = require('path');
+const util = require(`../app/utility`);
+var generators = require('yeoman-generator');
+var chalk = require('chalk');
+var yosay = require('yosay');
+var uuid = require('uuid');
+var ejs = require('ejs');
+
+function construct() {
+      // Calling the super constructor is important so our generator is correctly set up
+      generators.Base.apply(this, arguments);
+
+      this.argument('extName', { type: String, required: false, desc: 'Name of the extension' });
+}
+
+function input() {
+
+      // Collect any missing data from the user.
+      // This gives me access to the generator in the
+      // when callbacks of prompt
+      let cmdLnInput = this;
+
+      return this.prompt([
+            {
+                  type: 'input',
+                  name: 'extName',
+                  store: true,
+                  message: 'Please enter the (friendly) name of your extension:',
+                  validate: util.validateExtensionName,
+                  default: 'My Extension',
+                  when: answers => {
+                        return cmdLnInput.extName === undefined;
+                  }
+            }, {
+                  type: 'input',
+                  name: 'extId',
+                  store: true,
+                  message: 'Please enter the ID of your extension:',
+                  default: 'myextension',
+                  validate: util.validateExtensionId,
+                  when: answers => {
+                        return cmdLnInput.extId === undefined;
+                  }
+            }, {
+                  type: 'input',
+                  name: 'extDescription',
+                  store: true,
+                  message: 'Please enter a description of your extension:',
+                  default: 'My extension makes hubs great again',
+                  when: answers => {
+                        return cmdLnInput.extDescription === undefined;
+                  }
+            }, {
+                  type: 'input',
+                  name: 'publisherId',
+                  store: true,
+                  message: 'Please enter the name of your marketplace publisher ID:',
```

Comment:
uppercase "M" on Marketplace .. might even spell it out "Visual Studio Marketplace"